Terminal.Gui/Views/ListView.cs

```diff
@@ -807,32 +807,39 @@ public bool OnOpenSelectedItem ()
     }
 
     /// <inheritdoc/>
-    protected override bool OnKeyDown (Key a)
+    protected override bool OnKeyDown (Key key)
     {
         // If marking is enabled and the user presses the space key don't let CollectionNavigator
         // at it
         if (AllowsMarking)
         {
             var keys = KeyBindings.GetAllFromCommands (Command.Select);
 
-            if (keys.Contains (a))
+            if (keys.Contains (key))
             {
                 return false;
             }
 
             keys = KeyBindings.GetAllFromCommands ([Command.Select, Command.Down]);
 
-            if (keys.Contains (a))
+            if (keys.Contains (key))
             {
                 return false;
             }
 
         }
 
+        // If the key was bound to key command, let normal KeyDown processing happen. This enables overriding the default handling.
+        // See: https://github.com/gui-cs/Terminal.Gui/issues/3950#issuecomment-2807350939
+        if (KeyBindings.TryGet (key, out _))
```

Comment:
Is there any potential overlap with `Command.Select` and `Command.Down` used above vs the `AllowsMarking` flag to be false that could lead to weird behavior here? (Sorry I'm not knee-deep into this code so this might be a stupid question)

Reply:
Hmn good point, it actually looks like the entire `AllowsMarking` bit can go.

All it is doing is looking for specific keybindings to return `false` then we have a blanket if below it!

Good catch

Reply:
Fixed in [0e29e04](https://github.com/gui-cs/Terminal.Gui/pull/4029/commits/0e29e04c44a3725d8bd83fd5026f1a92e337944d)